src/Draco.Compiler/Internal/Documentation/DocumentationSection.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Draco.Compiler.Internal.Documentation;
+
+/// <summary>
+/// Represents a section of the documentation.
+/// </summary>
+/// <param name="Name">The name of the section</param>
+/// <param name="Elements">The <see cref="DocumentationElement"/>s this section contains.</param>
+internal sealed record class DocumentationSection(string Name, ImmutableArray<DocumentationElement> Elements)
```

Comment:
I feel like the design is flawed here. Name is not necessarily present, but section kind is. If the section kind annotates "other", no need to have some filler word for it.